eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -38,58 +48,167 @@ param (
-$purgeableResources = @(Get-PurgeableResources)
+function HasValidOwnerTag() {
+  if (!$ResourceGroup.Tags.Owners) {
+    return $false
+  }
+  $owners = $ResourceGroup.Tags.Owners -split "[;, ]"
+  $hasValidOwner = $false
+  $invalidOwners = @()
+  foreach ($owner in $owners) {
+      if (IsValidAlias -Alias $owner) {
+          $hasValidOwner = $true
+      } else {
+        $invalidOwners += $owner
+      }
+  }
+  if ($invalidOwners) {
+    Write-Warning " Resource group '$($ResourceGroup.ResourceGroupName)' has invalid owner tags: $($invalidOwners -join ',')"
+  }
+  if ($hasValidOwner) {
+      Write-Host " Skipping tagged exception resource group '$($ResourceGroup.ResourceGroupName)' with owners '$owners'"
+      return $true
+  }
+  return $false
+}
 
-foreach ($rg in $toDelete)
-{
-  if ($Force -or $PSCmdlet.ShouldProcess("$($rg.ResourceGroupName) (UTC: $($rg.Tags.DeleteAfter))", "Delete Group")) {
-    # Add purgeable resources that will be deleted with the resource group to the collection.
-    $purgeableResourcesFromRG = Get-PurgeableGroupResources $rg.ResourceGroupName
+function HasExpiredDeleteAfterTag([object]$ResourceGroup) {
+  $deleteDate = $ResourceGroup.Tags.DeleteAfter -as [DateTime]  # null if not exists
+  return $deleteDate -and [datetime]::UtcNow -gt $deleteDate
+}
 
-    if ($purgeableResourcesFromRG) {
-      $purgeableResources += $purgeableResourcesFromRG
-      Write-Verbose "Found $($purgeableResourcesFromRG.Count) potentially purgeable resources in resource group $($rg.ResourceGroupName)"
+function FindOrCreateDeleteAfterTag {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+  param(
+    [object]$ResourceGroup
+  )
+
+  if (!$ResourceGroup.Tags.DeleteAfter -as [datetime]) {
+    $deleteAfter = [datetime]::UtcNow.AddHours($DeleteAfterHours)
+    Write-Host "Adding DeleteAfer tag with value '$deleteAfter' to group '$(ResourceGroup.ResourceGroupName)'"
+    if ($Force -or $PSCmdlet.ShouldProcess("$($rg.ResourceGroupName) (UTC: $($rg.Tags.DeleteAfter))", "Update Group")) {
+      $ResourceGroup | Update-AzTag -Operation Merge -Tag @{ DeleteAfter = $deleteAfter }
     }
-    Write-Verbose "Deleting group: $($rg.ResourceGroupName)"
-    Write-Verbose "  tags $($rg.Tags | ConvertTo-Json -Compress)"
-    Write-Host ($rg | Remove-AzResourceGroup -Force -AsJob).Name
   }
 }
 
-# Purge all the purgeable resources and get a list of resources (as a collection) we need to follow-up on.
-Write-Host "Attempting to purge $($purgeableResources.Count) resources."
-$failedResources = @(Remove-PurgeableResources $purgeableResources -PassThru)
-if ($failedResources) {
-  Write-Warning "Timed out deleting the following $($failedResources.Count) resources. Please file an IcM ticket per resource type."
-  $failedResources | Sort-Object AzsdkResourceType, AzsdkName | Format-Table -Property @{l='Type'; e={$_.AzsdkResourceType}}, @{l='Name'; e={$_.AzsdkName}}
+function HasDeleteLock() {
```

Comment:
This looks redundant

Reply:
It's a placeholder right now but the logic will eventually be filled in as we start to support delete locks.